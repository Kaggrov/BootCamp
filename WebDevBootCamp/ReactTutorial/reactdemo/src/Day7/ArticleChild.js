import React from 'react'

const ArticleChild = () => {
  return (
    <div>
      <div className='App'>ArticleChild</div>
      <h1>Hello</h1>

    </div>
  )
}

export default ArticleChild