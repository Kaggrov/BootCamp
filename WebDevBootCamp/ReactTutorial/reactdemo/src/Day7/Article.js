import React, { useState } from 'react'
import ArticleChild from './ArticleChild.js'

const Article = () => {
    const [data,setData] = useState(0)
  return (
    <>
    <div className='App'>Article</div>
    <div>
        <ul>
            <li key="1" className='myCss'>C++</li>
            <li key="2" className='myCss'>C</li>
            <li key="3" className='myCss'>Java</li>
        </ul>
    </div>
    <div className='myCss2'>
        <span>1</span>
        <span>2</span>
        <span>3</span>
    </div>
    <button onClick={()=>setData(prev =>prev+1)}>Click Me</button>
    <p className={`click${data}`}>{data}</p>
    <ArticleChild/>
    </>
  )
}

export default Article