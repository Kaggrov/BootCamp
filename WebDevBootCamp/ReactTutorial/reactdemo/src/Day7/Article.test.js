import Enzyme ,{shallow} from 'enzyme'
import Article from './Article'
import ArticleChild from './ArticleChild'


describe('Article Component',()=>{
    test('render',()=>{

        let wrapper = shallow(<Article/>)
        expect(wrapper.exists()).toBe(true)
        console.log(wrapper.debug)
    })

    test('render2',()=>{

        let wrapper = shallow(<Article/>)
        expect(wrapper.find('.App').length).toEqual(1)
        console.log(wrapper.debug)
    })

    test('render3',()=>{

        let wrapper = shallow(<Article/>)
        expect(wrapper.contains(<div className='App'>Article</div>)).toBe(true)
        console.log(wrapper.debug)
    })

    test('render4',()=>{

        let wrapper = shallow(<Article/>)
        expect(wrapper.find('.myCss').at(0).key()).toBe("1")
        console.log(wrapper.debug)
    })

    test('test1',()=>{

        let wrapper = shallow(<Article/>)
        expect(wrapper.find('ul').children().length).toBe(3)
        console.log(wrapper.debug)
    })

    test('test2',()=>{

        let wrapper = shallow(<Article/>)
        expect(wrapper.find('.myCss2').childAt(0).type()).toBe('span')
        console.log(wrapper.debug)
    })

    // test('test3',()=>{

    //     let wrapper = shallow(<Article/>)
    //     wrapper.find('button').simulate('click')
    //     expect(wrapper.find('click1').length).toBe(1)
    //     console.log(wrapper.debug)
    // })

    test('test4',()=>{

        let wrapper = shallow(<Article/>)
        expect(wrapper.find('ArticleChild').dive().find('.App').length).toEqual(1);
        
        console.log(wrapper.debug())
    })

    test('test5',()=>{

        let wrapper = shallow(<ArticleChild/>)
        expect(wrapper).toMatchSnapshot();//to monitor changes in component for testing
    })
})