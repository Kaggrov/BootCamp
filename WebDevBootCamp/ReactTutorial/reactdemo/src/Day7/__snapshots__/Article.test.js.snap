// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component test5 1`] = `
<div>
  <div
    className="App"
  >
    ArticleChild
  </div>
  <h1>
    Hello
  </h1>
</div>
`;
